### DISCUSSION TITLE
Not sure why MDX stopped working all of a sudden?

### DISCUSSION THREAD

**Discussion by fableforce** [vercel/next.js#36848] [Help]
Hi all, hope you are well. 

I followed the doc's instructions to set up MDX with my site, and it worked on first try.

However, we were reviewing our code today, and realised:

1. Routes were still intact
2. However, we are being redirected to 404 when loading that page (while being on the correct route)

Not sure what I did wrong, but perhaps something to do with next.config (since I've added gsap support since then). Could someone kindly help me out? Thank you.

`next config:
`/** @type {import('next').NextConfig} */

const nextConfig = {
reactStrictMode: true,
};

const withMDX = require("@next/mdx")({
extension: /.mdx?$/,
options: {
remarkPlugins: [],
rehypePlugins: [],
// If you use MDXProvider, uncomment the following line.
// providerImportSource: "@mdx-js/react",
},
});

// Let next work with transpile modules for GSAP.
const withTM = require("next-transpile-modules")(["gsap"]);

module.exports = { nextConfig }; // module exports can be an object

module.exports = withMDX({
// Append the default value with md extensions
pageExtensions: ["ts", "tsx", "js", "jsx", "md", "mdx"],
images: {
domains: ["images.unsplash.com", "tailwindui.com"],
},
});

module.exports = withTM();
`

specific mdx file
`import LegalLayout from "components/layout/LegalLayout";

export const meta = {
author: "nil",
title: "Cookie policy",
lastUpdated: "28th April, 2022",
};

## Content here
export default ({ children }) => (
{children}
);
`
### layout file
`import Head from "next/head";

export default function LegalLayout({ meta, children }) {
return (
<>
<title>Create Next App</title>
{/* Hero section */}
{meta.title}
{meta.lastUpdated}
    {/* Article starts here. */}
    <article className="mx-auto prose max-w-prose">
      <div>{children}</div>
      <div>{meta.author}</div>
    </article>
  </main>
</>
);
}
`

**Comment by icyJoseph** [comment]
So the `next.config.js` looks like this?

```ts
/** @type {import('next').NextConfig} */
const nextConfig = {
  reactStrictMode: true,
};

const withMDX = require("@next/mdx")({
  extension: /.mdx?$/,
  options: {
    remarkPlugins: [],
    rehypePlugins: [],
    // If you use MDXProvider, uncomment the following line.
    // providerImportSource: "@mdx-js/react",
  },
});

// Let next work with transpile modules for GSAP.
const withTM = require("next-transpile-modules")(["gsap"]);

module.exports = { nextConfig }; // module exports can be an object

module.exports = withMDX({
  // Append the default value with md extensions
  pageExtensions: ["ts", "tsx", "js", "jsx", "md", "mdx"],
  images: {
    domains: ["images.unsplash.com", "tailwindui.com"],
  },
});

module.exports = withTM();
```

I might be wrong, but I think the issue is that you override the `module.exports` a few times. What you need to do is compose the different Next.js configs.

```js
/** @type {import('next').NextConfig} */
const nextConfig = {
  reactStrictMode: true,
  pageExtensions: ["ts", "tsx", "js", "jsx", "md", "mdx"],
  images: {
    domains: ["images.unsplash.com", "tailwindui.com"],
  },
};

const withMDX = require("@next/mdx")({
  extension: /.mdx?$/,
  options: {
    remarkPlugins: [],
    rehypePlugins: [],
    // If you use MDXProvider, uncomment the following line.
    // providerImportSource: "@mdx-js/react",
  },
});

const withTM = require("next-transpile-modules")(["gsap"]);

module.exports = withTM(withMDX(nextConfig));
```
